pypuppetdb/api/__init__.py

```diff
@@ -312,10 +313,22 @@ def _query(self, endpoint, path=None, query=None,
             payload = None
 
         try:
-            r = self._session.get(url, params=payload, verify=self.ssl_verify,
-                                  cert=(self.ssl_cert, self.ssl_key),
-                                  timeout=self.timeout,
-                                  auth=(self.username, self.password))
+            if request_method.upper() == 'GET':
+                r = self._session.get(url, params=payload,
+                                      verify=self.ssl_verify,
+                                      cert=(self.ssl_cert, self.ssl_key),
+                                      timeout=self.timeout,
+                                      auth=(self.username, self.password))
+            elif request_method.upper() == 'POST':
+                r = self._session.post(url, params=payload,
+                                       verify=self.ssl_verify,
+                                       cert=(self.ssl_cert, self.ssl_key),
+                                       timeout=self.timeout,
+                                       auth=(self.username, self.password))
+            else:
+                log.error("Only GET or POST supported, {0} unsupported".format(
+                          request_method))
+                raise APIError
```

Comment:
None of the other pypuppetdb exceptions raised here pass or accept error messages.